### PR TITLE
V1.23.0 stage

### PR DESCRIPTION
**Pull request by ubhattacharjya**
### Summary

ECS Agent Staging v1.23.0

### Testing

- [ ] Builds on Linux (`make release`)
- [ ] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [ ] Unit tests on Linux (`make test`) pass
- [ ] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [ ] Integration tests on Linux (`make run-integ-tests`) pass
- [ ] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [ ] Functional tests on Linux (`make run-functional-tests`) pass
- [ ] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+## 1.23.0
+* Feature - Add support for ECS Secrets integrating with AWS Secrets Manager [#1713](https://github.com/aws/amazon-ecs-agent/pull/1713)
+* Enhancement - Add availability zone to task metadata endpoint and ECS metadata file
```
Comment (adnxn) [inline CHANGELOG.md]: pr link for this?

Comment (ubhattacharjya) [reply CHANGELOG.md]: Added the PR links

File: VERSION [hunk 2]
```diff
@@ -1 +1 @@
-1.22.0
+1.23.0
```
Comment (yumex93) [inline VERSION]: There should be an empty line in the end.

**Comment by petderek** [comment]
Can we sneak in the functional test changes too? #1723

File: CHANGELOG.md (hunk 1 above)
Comment (haikuoliu) [reply CHANGELOG.md]: wait i didn't put pr link because this feature was done by 2 prs

Comment (haikuoliu) [reply CHANGELOG.md]: ok nvm I saw your update

**Pull request closed without merging**